File: coding-test/programmers/src/kr/co/programmers/practice/SumOfFactors.java
package kr.co.programmers.practice;

public class SumOfFactors {

    public static void main(String[] args) {
        int n = 12;
        System.out.println(new Solution().solution(n));
    }

    static class Solution {
        public int solution(int n) {
            int answer = 0;

            for(int i = 1; i <= n; i++){
                if ( n % i == 0 ) {
                    answer += i;
                }
            }
            return answer;
        }
    }
}

File: coding-test/nodejs/src/practice/sum-of-factors.js
// @ts-check

/**
 *
 * @param {number} n
 */
function solution(n) {
  let answer = 0

  for (let i = 1; i <= Math.floor(Math.sqrt(n)); i += 1) {
    if (n % i === 0) {
      if (i === Math.sqrt(n)) {
        answer += i
      } else {
        answer += i
        answer += n / i
      }
    }
  }

  return answer
}

console.log(solution(12))
